### DISCUSSION TITLE
Supabase in Chrome Extension MV3 Service Workers

### DISCUSSION THREAD

**Discussion by karol-cf** [supabase/supabase#6527] [Questions]
Hi,

I'm going to describe the issue with an example of an extension, I've found in [this YouTube video](https://www.youtube.com/watch?v=SP7eM_nXgJ4) (he's using MV2, which utilizes a background page, instead of service workers, so the issue is not present there). The extension allows you to sign up and log in with Supabase in a new Chrome tab. I've got a bundled version of supabase-js in my project. The supabase-js library is imported in the service worker and I create a client (as you can see on the screenshot below, I've tried binding the fetch as well). Whenever the user wants to authenticate, I send a message through the message passing API to my service worker and call various supabase.auth methods. Every time the signUp method is called, for example, I get "ReferenceError: XMLHttpRequest is not defined" and the method does not proceed further (on the third screenshot I've tried signing up and it threw an error).

![image](https://user-images.githubusercontent.com/90687732/164671121-7ad79d14-9391-4db5-b0fd-284d7b303a30.png)
![image (1)](https://user-images.githubusercontent.com/90687732/164671261-4ac1f4c1-650c-4b8c-b0c0-2465fd053f00.png)
![image (2)](https://user-images.githubusercontent.com/90687732/164671390-7c7af7c8-d02f-4980-a734-d3e2d1364343.png)

I've read about this issue [here](https://github.com/lquixada/cross-fetch/issues/78), but it wasn't really helpful. The only solution, that I think might work is removing the cross-fetch import from your lib and then bundling it. Because of the MV3 requirements, I need that bundled js file anyway. However, I'd much rather not do that, because I don't think that this is a clean way of solving the problem. Any ideas on how to solve it? (I'm going to add this comment to that cross-fetch issue as well)

**Comment by edwinsjr** [comment]
Any update on this? It is still an issue for us.

**Comment by MatthewLee72** [comment]
bump

**Comment by dantheman0207** [comment]
This is still relevant to me, would love some guidance here.

**Comment by shaezzy** [comment]
What version of the js sdk are you using? In most recent version, fetch override should be in global:

```js
 const supabase = createClient(
    process.env.SUPABASE_URL!,
    process.env.SUPABASE_ANON_KEY!,
    {
      auth: {
        flowType: "pkce",
      },
      global: {
        fetch: fetch,
      },
    }
  );

**Comment by sorokinvj** [comment]
Has anybody succeeded with integration of supabase-js in service worker with manifest v3? I am considering moving away from Firebase/Firestore
